### PR TITLE
Don't wait after the final copy attempt

### PR DESCRIPTION
**Pull request by kewilli**
Without these changes:
* Try to copy from all locations. If any succeed, return
* Wait, then loop

This leads to us waiting after the final attempt.

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs [hunk 1]
```diff
@@ -169,17 +183,6 @@ protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
...
-                    long waitTicks = _retryIntervals[attemptCount].Ticks;
-
-                    // Randomize the wait delay to `[0.5 * delay, 1.5 * delay)`
-                    TimeSpan waitDelay = TimeSpan.FromTicks((long)((waitTicks / 2) + (waitTicks * ThreadSafeRandom.Generator.NextDouble())));
-
-                    Tracer.Warning(operationContext, $"{AttemptTracePrefix(attemptCount)} All replicas {hashInfo.Locations.Count} failed. Retrying for hash {hashInfo.ContentHash.ToShortString()} in {waitDelay.TotalMilliseconds}ms...");
-
-                    attemptCount++;
-
-                    await Task.Delay(waitDelay, cts);
```
Comment (SergeyTeplyakov) [inline Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs]: if (attemptCount != _retryIntervals.Count) {
  await Task.Delay();
}

Comment (SergeyTeplyakov) [reply Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs]: And the rest stays the same

---
In reply to: [314512295](https://github.com/microsoft/BuildXL/pull/743#discussion_r314512295) [](ancestors = 314512295)

**Comment by dcwuser** [comment]
>                         break;

Log something like "success at attempt {attemptCount}".

---
Refers to: Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs:152 in 80c78a8. [](commit_id = 80c78a8540c887267c3d735688ddfa1229baf8ef, deletion_comment = False)

**Comment by dcwuser** [comment]
>                 while (attemptCount < _retryIntervals.Count && (putResult == null || !putResult))

Willing to live with this, but still suggest while (true) is cleaner now.

---
Refers to: Public/Src/Cache/ContentStore/Distributed/Stores/DistributedContentCopier.cs:137 in 80c78a8. [](commit_id = 80c78a8540c887267c3d735688ddfa1229baf8ef, deletion_comment = False)

**Pull request merged**